### PR TITLE
feat: add explorer's link and contract name when approving

### LINKED ISSUE #2268: Showing name + linking to approval contract on approval step
### Overview

Gm!

Currently we do not allow users to see what contract is requesting they approve spend access on the approval step.

181080714-d2956380-fc0d-4d3c-a2ce-34319e9b7632

We simply want to allow users to click and open a new tab to the contract in Etherscan.

Desired text on approval screen should be as follows.
_[name] needs your permission to use your [asset]_

Where [name] is the contract name and if possible, a link to the contract in Etherscan. If the [name] field cannot be both, we can link to it elsewhere - please ping me in that scenario for a workaround.

This ticket aims to close #2240 from the great @gomesalexandre 

### References and additional details

* Mock `DisplayFeeData` where the `tradeFeeSource` can be seen:

https://github.com/shapeshift/web/blob/9a7c5095caf25fa2cdb4ef46a30013a42e98566e/src/test/constants.ts#L104

* `<Approval />` component where the `tradeFeeSource` can be gotten from `fees.tradeFeeSource`:

https://github.com/shapeshift/web/blob/222fb5c40ec01e7687254ffdf31e40476efbf8a6/src/components/Approval/Approval.tsx#L51

### Acceptance Criteria

- User can see the name of the contract theyre interacting with (0x, CowSwap, etc)
- User can link out to the contract theyre interacting with in Etherscan

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

1 hour

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description
I still added the contract name inside a link to the explorer.

There are 2 approvals components, `Approval.tsx` is linked to the swap feature, this one is afaik only linked to ERC20, meaning that we can use etherscan

The other one (`Approve.tsx`) is linked to DeFi features, which will be linked to Cosmos soon and other blockchain we will be supporting some staking contracts

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)
closes #2268

## Testing
- Try to swap a token, like FOX => ETH for example, when you see the approval window, you'll see the swapper name + a link to the explorer 
- Try to stake some Fox to the Foxy contract, you'll see the approval modal with the same thing

## Screenshots (if applicable)
image
image

### CODE REVIEW

File: src/features/defi/components/Approve/Approve.tsx [hunk 1]
```diff
@@ -79,7 +81,10 @@ export const Approve = ({
         </Stack>
         <Stack>
           <Text fontWeight='bold' translation={['modals.approve.header', { asset: asset.name }]} />
-          <Text color='gray.500' translation={['modals.approve.body', { asset: asset.name }]} />
+          <CText color='gray.500'>
+            <Link href={`https://etherscan.io/address/${contractAddress}`}>Contract</Link>
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/features/defi/components/Approve/Approve.tsx:85]: Fixed as seen together, I used `ShapeShift DAO Router` as used everywhere else on the DeFi part

File: src/features/defi/components/Approve/Approve.tsx [hunk 2]
```diff
@@ -79,7 +81,17 @@ export const Approve = ({
...
           <Text fontWeight='bold' translation={['modals.approve.header', { asset: asset.name }]} />
-          <Text color='gray.500' translation={['modals.approve.body', { asset: asset.name }]} />
+          <CText color='gray.500'>
+            <Link
+              href={`${asset.explorerAddressLink}${contractAddress}`}
+              color='blue.500'
+              me={1}
+              isExternal
+            >
+              {translate('modals.approve.routerName')}
```
Comment (gomesalexandre) [inline src/features/defi/components/Approve/Approve.tsx:91]: I can kind of see why the original `Allow ShapeShift DAO Router to use your %{asset}` was there as a string, but would prefer this to be dynamic now, passed as a prop to the component, and then interpolated.

Given the ramifications of working with the DeFi UI abstraction and the diff size, this doesn't need to be as part of this PR but would be a nice-to-have refactor follow-up

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Approve.tsx [hunk 3]
```diff
@@ -157,6 +157,7 @@ export const Approve: React.FC<YearnApproveProps> = ({ onNext }) => {
       learnMoreLink='https://shapeshift.zendesk.com/hc/en-us/articles/360018501700'
       onCancel={() => onNext(DefiStep.Info)}
       onConfirm={handleApprove}
+      contractAddress={contractAddress}
```
Comment (gomesalexandre) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Approve.tsx:160]: `contractAddress` is the address of the token being staked, not the address of the actual contract. What you're looking for is `ssRouterContractAddress` from `investor-yearn` - it is the same ShapeShift router contract for all Yearn tokens stakings

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Tested locally, FOXy Deposit/Withdraw and swapper approvals are showing the right name and linking to the right contract. 

For Yearn (yvUSDT vault) this is properly displaying the Shapeshift DAO router name and and prompting to sign a Tx `approve()`ing it, but is linking to a different contract address (the token i.e [USDT yVault (yvUSDT)](https://etherscan.io/token/0x7da96a3891add058ada2e826306d812c638d87a7))
image

File: src/features/defi/providers/foxy/components/FoxyManager/Deposit/FoxyDeposit.tsx [hunk 4]
```diff
@@ -95,6 +95,9 @@ export const FoxyDeposit = () => {
       [DefiStep.Approve]: {
         label: translate('defi.steps.approve.title'),
         component: Approve,
+        props: {
+          contractAddress,
+        },
```
Comment (gomesalexandre) [inline src/features/defi/providers/foxy/components/FoxyManager/Deposit/FoxyDeposit.tsx:100]: Not a fan of a catch-all prop called props, especially not typed as `Record<any, any>` but that works for the purpose of fixing this

**Review by gomesalexandre — APPROVED** [review]
Tested again with the previous testing flow, works as previously tested and is now linking to the Shapeshift DAO Yearn Router for Yearn

**Pull request merged**